sentry-android-core/src/test/java/io/sentry/android/core/internal/util/PrivilegeEscalationViaContentProviderCheckerTest.kt

```diff
@@ -0,0 +1,105 @@
+package io.sentry.android.core.internal.util
+
+import android.content.ContentProvider
+import android.content.Context
+import android.os.Build
+import io.sentry.android.core.BuildInfoProvider
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verifyNoInteractions
+import org.mockito.kotlin.whenever
+import kotlin.test.Test
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+class PrivilegeEscalationViaContentProviderCheckerTest {
+
+    private class Fixture {
+        val buildInfoProvider = mock<BuildInfoProvider>()
+
+        fun getSut(
+            sdkVersion: Int = Build.VERSION_CODES.O
+        ): PrivilegeEscalationViaContentProviderChecker {
+            whenever(buildInfoProvider.sdkInfoVersion).thenReturn(sdkVersion)
+
+            return PrivilegeEscalationViaContentProviderChecker(buildInfoProvider)
+        }
+    }
+
+    private val fixture = Fixture()
+
+    @Test
+    fun `When sdk version is less than vulnerable versions, security check is not performed`() {
+        val contentProvider = mock<ContentProvider>()
+
+        fixture.getSut(Build.VERSION_CODES.N_MR1).securityCheck(contentProvider)
+
+        verifyNoInteractions(contentProvider)
+    }
+
+    @Test
+    fun `When sdk version is greater than vulnerable versions, security check is not performed`() {
+        val contentProvider = mock<ContentProvider>()
+
+        fixture.getSut(Build.VERSION_CODES.Q).securityCheck(contentProvider)
+
+        verifyNoInteractions(contentProvider)
+    }
+
+    @Test
+    fun `When calling package is null, security check exception is thrown`() {
+        val contentProvider = mock<ContentProvider>()
+
+        contentProvider.mockPackages(null)
+
+        var securityException: SecurityException? = null
```

Comment:
I think you can simplify this by using assertThrows
```suggestion
        assertThrows<SecurityException> { fixture.getSut().securityCheck(contentProvider) }
```

Reply:
Great callout! I'm an idiot 🤣 I'll change 🔥

Reply:
Hmm... Wait a sec... I know I initially tried `assertThrows` but no code suggestion was popping up for it. 

Looking for all instances in java or Kotlin files...

<img width="1316" alt="Screen Shot 2023-01-13 at 7 22 20 AM" src="https://user-images.githubusercontent.com/4332753/212380771-e6a06842-8835-47f3-95f3-b5f828873946.png">

Seems like it requires a dependency currently only used in `sentry-spring` and `sentry-spring-jakarta`.

<img width="524" alt="Screen Shot 2023-01-13 at 7 25 39 AM" src="https://user-images.githubusercontent.com/4332753/212381336-9c16e247-dd93-4706-a5ba-8c99fee2f2ff.png">

I'd like to avoid changing any build configurations in this PR and since I'm only using this `try-catch` approach in two places in this one file, maybe we can let this slide? If you feel strongly about using `assertThrows`, you can make changes after the PR has been merged 🙏